app/code/Magento/Search/view/frontend/web/js/form-mini.js

```diff
@@ -338,6 +339,11 @@ define([
                                     this._resetResponseList(false);
                                 }
                             }.bind(this));
+                    } else {
```

Comment:
I would like to avoid code duplication  between line 343 and line 350.
Do you think creating a dedicated method to close the autocompletition menu would be a good idea?

Reply:
What type of method? This would probably require a refactor as this code is run in different situations and at specific positions in the code.

Reply:
Applies to other code parts too.

For example

```
this._getFirstVisibleElement().addClass(this.options.selectClass);
this.responseList.selected = this._getFirstVisibleElement();
```

Let's improve this in another PR to avoid issues with `this` and to not add any new methods to 2.2 so far.

Reply:
It makes sense. LGTM